lib/finch/conn.ex

```diff
@@ -25,32 +24,28 @@ defmodule Finch.Conn do
     }
 
     Telemetry.event(:reused_connection, %{}, meta)
-    conn
+    {:ok, conn}
   end
 
-  def connect(conn), do: connect(conn, conn.backoff.initial)
-
-  def connect(conn, prev_backoff) do
+  def connect(conn) do
     meta = %{
       scheme: conn.scheme,
       host: conn.host,
       port: conn.port
     }
 
     start_time = Telemetry.start(:connect, meta)
+    conn_opts = Keyword.merge([mode: :passive], conn.opts)
 
-    with {:ok, mint} <- HTTP.connect(conn.scheme, conn.host, conn.port, conn.opts),
+    with {:ok, mint} <- HTTP.connect(conn.scheme, conn.host, conn.port, conn_opts),
          {:ok, mint} <- HTTP.controlling_process(mint, conn.parent) do
```

Comment:
WWe no longer need this line, right? Since we are now connecting from the process that is going to use it?

Reply:
Ah good, point!